### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by serhido**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,45 @@
...
+            [
+                {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["text"],
+                "source": "flake8"
+                }
+                for error in errors
+            ],
```
Comment (Vasyl-Smutok) [inline app/main.py:22]: use `format_linter_error` here

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,45 @@
...
+        "path": linter_report[i],
+        "status": "passed" if "errors" == [] else "failed"
+        }
```
Comment (Vasyl-Smutok) [inline app/main.py:43]: use `format_single_linter_file` here

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
+            }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(linter_report[i])
+        for i, errors in enumerate(linter_report)
+    ]
```
Comment (Vasyl-Smutok) [inline app/main.py:26]: `i` bad variable name

File: app/main.py [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
-    pass
+    return {
+        "errors":
+            [
+                format_linter_error(error)
+                for error in errors
+            ],
+        "path": file_path,
+        "status": "passed" if "errors" == [] else "failed"
+            }
```
Comment (Vasyl-Smutok) [inline app/main.py:19]:
```suggestion
return {
    "errors": [format_linter_error(error) for error in errors],
    "path": file_path,
    "status": "failed" if errors else "passed""
}
```

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
+    }
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors":[format_linter_error(error)for error in errors],
+        "path": file_path,
+        "status": "passed" if "errors" == [] else "failed"
+            }
```
Comment (ArsenPidhoretskyi) [inline app/main.py:15]:
```suggestion
    }
```

File: app/main.py [hunk 6]
```diff
@@ -1,13 +1,22 @@
...
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(linter_report[count])
+        for count, errors in enumerate(linter_report)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:21]: why you need enumerate
use .items method

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**